srml/support/src/error.rs

```diff
@@ -0,0 +1,205 @@
+#[macro_export]
+macro_rules! impl_outer_error {
```

Comment:
Do we really need this? First I don't want to like to have the user give `Error` to `construct_runtime!` for each module. Secondly, why can we not just use `DispatchableError` in `Dispatchable`.

Reply:
I was just follow how events is implemented. But you are right, I will try just use `DispatchableError`

Reply:
We still need a runtime Error type to bridge module Error and DispatchError. But the implementation can be simplified.